test/app.test.js

```diff
@@ -1,10 +1,250 @@
-import { getNearestShops } from '../src/app';
-
-describe('App', () => {
-  it('should return an array when the input is valid', () => {
-    expect(Array.isArray(getNearestShops({
-      lat: 0,
-      lng: 0,
-    }))).toBe(true);
+import { getNearestShops } from "../src/app";
+import DeltaCoffeeShop from "../src/deltaCoffeeShop";
+import DeltaCoffeeShopList from "../src/deltaCoffeeShopList";
+import SortedDeltaCoffeShopList from "../src/sortedDeltaCoffeeShopList";
+import errorCodes from "../src/errorcodes.js";
+import { ERROR_TOKEN, errorResponse } from "../src/utils.js";
+
+describe("App", () => {
+  it("should return an array when the input is valid", async () => {
+    const f = await getNearestShops({ x: 0, y: 0 });
+    expect(Array.isArray(f)).toBe(true);
+  });
+
+  it("should return an array even if the input is invalid", async () => {
+    const f = await getNearestShops({ x: 0, y: "abc" });
+    expect(Array.isArray(f)).toBe(true);
+  });
+
+  it("should return an array with the error code if the input is invalid", async () => {
+    const f = await getNearestShops({ x: 0, y: "abc" });
+    expect(f.length == 2).toBe(true);
+    expect(f[0]).toBe(ERROR_TOKEN);
+    expect(f[1]).toBe(errorCodes.INVALID_INPUT_ARGS_NAN);
+  });
+
+  it("should return an array with three coffee shops, containing the name and the delta value", async () => {
+    const f = await getNearestShops({ x: 47.6, y: -122.4 });
+    expect(f.length == 3).toBe(true);
+    f.forEach(function (coffeeShop) {
+      expect(coffeeShop[0] === ERROR_TOKEN).toBe(false);
+      expect(typeof coffeeShop.name).toBe("string");
+      expect(typeof coffeeShop.delta).toBe("number");
+    });
+  });
+});
+
+describe("Utils", () => {
+  it("should return the error response in the expected format", () => {
+    let resp = errorResponse(errorCodes.GENERIC_ERROR);
```

Comment:
It is inconvenient for the reader to follow a test with this many assertions. Also using let induces people into thinking that this will change later in the code, making it's value unreliable. How can this test be improved?

Reply:
Instead we could iterate over the defined errorcodes and make this check for all of them.   

for (const errorCode in errorCodes) {
   &nbsp;&nbsp;const resp = errorResponse(errorCode);
   &nbsp;&nbsp;const expectedResp = [ERROR_TOKEN, errorCode];
   &nbsp;&nbsp;expect(resp[0]).toBe(expectedResp[0]);
   &nbsp;&nbsp;expect(resp[1]).toBe(expectedResp[1]);
    }